pyblish/lib.py

```diff
@@ -349,3 +351,27 @@ def _resolve_name(name, package, level):
     __import__(name)
 
     return sys.modules[name]
+
+
+def emit(signal, **kwargs):
+    """Searches through the registered callbacks and executes any callback that
+    matches the signal.
+    The keyword arguments (kwargs) are passed directly through from the emitter
+    to the callbacks.
+
+        Arguments:
+            signal (string): Name of signal emitted
+
+        Example:
```

Comment:
For example:

```python
"""
Example:
    >>> import sys
    >>> from .plugin import register_callback
    >>> register_callback("mysignal", lambda data: sys.stdout.write(data))
    >>> emit("mysignal", data={"something": "cool"})
    {'something': 'cool'}
"""
```

You can test the test, with `doctest.testmod()` from another file or terminal.

```python
>>> import doctest
>>> import pyblish.lib
>>> doctest.testmod(pyblish.lib)
```